lib/init.lua

```diff
@@ -471,7 +471,11 @@ function Promise.is(object)
 		return false
 	end
 
-	return type(object.andThen) == "function"
+	local ok, isPromise = pcall(function()
+		return type(object.andThen) == "function"
+	end)
+
+	return ok and isPromise
```

Comment:
```suggestion
	local objectMetatable = getmetatable(object)
	return type(objectMetatable) == "table" and type(rawget(objectMetatable, "andThen")) == "function"
```

Reply:
How does this look to you?

Reply:
That should work, and I'm not seeing any other potential edge cases left behind:

- [x] If `__metatable` is set, the first check will fail and so return false
- [x] If otherwise, the "andThen" function will be checked via `rawget` which will bypass the `__index` metamethod.

Probably there might be some code cleaning that might be done before actually merging it (e.g. `objectMetatable == nil` probably could be `not objectMetatable`, you know, nitpicks) - do you want me to do it or do you prefer doing it yourself?

Reply:
I think `== nil` is clearer in this case

Reply:
> EDIT: There might be one, and that is "what if the object is not a Promise but for some reason has an unrelated andThen method?" (I know, though, that this is a long shot)

This is just the way it works. andThen indicates chainability, so if that is the case then the Promise library will chain onto *whatever* that thing is, even if it's not from this library. 

This is also the case in JavaScript promises. They will attempt to chain into any object with a `then` method.